crates/text_edit/src/lib.rs

```diff
@@ -16,6 +18,22 @@ pub struct Indel {
     pub delete: TextRange,
 }
 
+impl PartialOrd for Indel {
+    fn partial_cmp(&self, other: &Indel) -> Option<Ordering> {
+        if self == other {
+            return Some(Ordering::Equal);
+        }
+        let cmp_to_other = self.delete.end() <= other.delete.start();
+        let cmp_to_self = other.delete.end() <= self.delete.start();
+        match (cmp_to_other, cmp_to_self) {
+            (true, true) => None,
+            (true, false) => Some(Ordering::Less),
+            (false, true) => Some(Ordering::Greater),
+            (false, false) => None,
+        }
```

Comment:
comparing `self.delete.end()` with `other.delete.start()` should be mostly sufficient, only requiring extra checks in case those offsets are equal I believe

Reply:
I don't think so, because we have to make sure that `self > other` is true if and only if `other < self`. So we have to check both directions

Reply:
Well it depends on what we want the ordering to be like for a range that is included in another.
That is for 1..5 and 2..4, what do we want the ordering to be like.
The previous behaviour was just checking overlap via `l.delete.end() <= r.delete.start() || l == r`(l being self, r being other) which ignored `r.delete.end()` and `l.delete.start()`.

On that note it might not really make sense to implement `PartialOrd` here, and instead just use a function that checks the one comparison we want given that there isn't really an Ordering you can assign to ranges per se.

Reply:
@Veykril Yeah, that makes sense